cloud/ec2.go

```diff
@@ -613,6 +616,30 @@ func (m *ec2Manager) extendExpiration(ctx context.Context, h *host.Host, extensi
 	return errors.Wrapf(h.SetExpirationTime(h.ExpirationTime.Add(extension)), "error extending expiration time in db for '%s'", h.Id)
 }
 
+func (m *ec2Manager) attachVolume(ctx context.Context, h *host.Host, volume string) error {
```

Comment:
Are we checking somewhere that the volume isn't attached to another host? Better not to rely on AWS for this.

Reply:
Made a FindHostWithVolume function in the host package that takes a volume ID. Added a check at the host modify rest route that the volume is not already attached to a host.

Reply:
Although I'm now realizing that querying just on the volumeIDs will also look through terminated hosts -- need to investigate what happens to volumes when hosts are terminated and if we should remove volumeIDs from hosts at termination.

Reply:
Decided to remove volumes from host document at termination -- problem should be solved.